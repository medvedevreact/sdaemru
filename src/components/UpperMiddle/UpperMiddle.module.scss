.upperMiddle {
  padding: 0 40px;
  height: 40px;
  display: flex;
  align-items: center;
  li {
    cursor: pointer;
  }
}
.list {
  display: flex;
  list-style-type: none;
  gap: 75px;
  opacity: 0.75;
}
