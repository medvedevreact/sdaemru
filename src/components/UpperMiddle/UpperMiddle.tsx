import styles from "./UpperMiddle.module.scss";

const buttons = ["Мой профиль", "Добавить объявление"];

export const UpperMiddle = () => {
  return (
    <div className={styles.upperMiddle}>
      <ul className={styles.list}>
        {buttons.map((el) => (
          <li>{el}</li>
        ))}
      </ul>
    </div>
  );
};
