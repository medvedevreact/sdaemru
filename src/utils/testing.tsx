import { MemoryRouter, MemoryRouterProps } from "react-router-dom";
import { render, RenderOptions } from "@testing-library/react";
import { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import { store } from "../store";

interface RenderWithRouterOptions
  extends MemoryRouterProps,
    Omit<RenderOptions, "wrapper"> {}

export const renderWithRouter = (
  component: ReactElement,
  options: RenderWithRouterOptions = {}
) => {
  const Wrapper = ({ children }: { children?: ReactNode }) => (
    <MemoryRouter {...options}>
      <Provider store={store}>{children}</Provider>
    </MemoryRouter>
  );

  return render(component, { wrapper: Wrapper, ...options });
};
