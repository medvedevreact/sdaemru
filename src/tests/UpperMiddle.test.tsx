import { screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { renderWithRouter } from "../utils/testing";
import { UpperMiddle } from "../components/UpperMiddle/UpperMiddle";
import userEvent from "@testing-library/user-event";
import { useNavigate } from "react-router-dom";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: jest.fn(),
}));

const useNavigateMock = jest.mocked(useNavigate);

describe("upper middle component", () => {
  test("render", () => {
    renderWithRouter(<UpperMiddle />);
    expect(expect(screen.getAllByRole("listitem")).toMatchSnapshot());
  });
  test("navigate to other page", async () => {
    const navigateMock = jest.fn();

    useNavigateMock.mockReturnValue(navigateMock);
    renderWithRouter(<UpperMiddle />);
    const button = screen.getByText(/Добавить объявление/i);
    await userEvent.click(button);

    expect(navigateMock).toHaveBeenCalledTimes(1);
  });
});
