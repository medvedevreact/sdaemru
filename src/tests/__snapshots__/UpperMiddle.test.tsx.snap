// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`upper middle component render 1`] = `
[
  <li>
    Мой профиль
  </li>,
  <li>
    Добавить объявление
  </li>,
]
`;
